bots/mirror/src/main/java/org/openjdk/skara/bots/mirror/MirrorBot.java

```diff
@@ -79,13 +79,24 @@ public Collection<WorkItem> run(Path scratchPath) {
             var sanitizedUrl =
                 URLEncoder.encode(to.webUrl().toString(), StandardCharsets.UTF_8);
             var dir = storage.resolve(sanitizedUrl);
+            var dir_temporary = storage.resolve(sanitizedUrl + "temporary");
             Repository repo = null;
 
 
             if (!Files.exists(dir)) {
+                if (Files.exists(dir_temporary)) {
+                    try {
+                        Files.walk(dir_temporary)
+                                .map(Path::toFile)
+                                .sorted(Comparator.reverseOrder())
+                                .forEach(File::delete);
+                    } catch (IOException io) {
+                        throw new RuntimeException(io);
+                    }
+                }
                 log.info("Cloning " + from.name());
-                Files.createDirectories(dir);
-                repo = Repository.mirror(from.url(), dir);
+                Files.createDirectories(dir_temporary);
+                repo = Repository.mirror(from.url(), dir_temporary);
             } else {
                 log.info("Found existing scratch directory for " + to.name());
                 repo = Repository.get(dir).orElseThrow(() -> {
```

Comment:
Instead of introducing a temporary directory, I would recommend just falling back on recloning if `Repository::get` fails. Something like this:
```
                repo = Repository.get(dir).orElseGet(() -> {
                    try {
                        Files.walk(dir_temporary)
                                .map(Path::toFile)
                                .sorted(Comparator.reverseOrder())
                                .forEach(File::delete);
                    } catch (IOException e) {
                        throw new UncheckedIOException(e);
                    }
                    Repository.mirror(from.url(), dir);
                );
```

Reply:
Yes, it's much better! Thank you!